tests/rados/test_mon_connection_scores.py

```diff
@@ -0,0 +1,184 @@
+"""
+Module to verify connection scores of monitor dameons are valid
+at different scenarios
+"""
+
+import random
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.monitor_workflows import MonitorWorkflows
+from tests.rados.monitor_configurations import MonElectionStrategies
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    # CEPH-83602911
+    Covers:
+        - BZ-1892173
+        - BZ-2174954
+        - BZ-2151501
+        - BZ-2315595
+        - BZ-2326179
+    Module to verify connection scores of monitor
+    dameons are valid at different scenarios
+    Returns:
+        1 -> Fail, 0 -> Pass
+    #steps
+    1. Deploy a Ceph cluster
+    2. Validate correct number of mons in connection score,
+       correct number of peers for each mon, each mon present in quorum
+       and valid connection score ( score between 0 - 1 )
+    3. Change election strategy to disallow mode and perform
+       validations mentioned in step(2)
+    4. Change election strategy to connectivity mode and perform
+       validations mentioned in step(2)
+    5. Add new host as mon and perform validations
+       mentioned in step(2)
+    6. Remove mon added in step(5) in the cluster
+       and perform validation mentioned in step(2)
+    7. Restart monitor service using systemctl
+       and perform validations mentioned in step(2)
+    8. Stop monitor service using systemctl
+       and perform validations mentioned in step(2)
+    9. Start monitor service using systemctl
+       and validations mentioned in step(2)
+    """
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mon_election_obj = MonElectionStrategies(rados_obj=rados_obj)
+    mon_workflow_obj = MonitorWorkflows(node=cephadm)
+
+    try:
+        # Collect mon details such as name, rank and number of mons
+        mon_nodes = ceph_cluster.get_nodes(role="mon")
+        osd_nodes = ceph_cluster.get_nodes(role="osd")
+        mon_nums = len(mon_nodes)
+
+        # Election strategy is classic by default, connection score validation
+        if not mon_workflow_obj.connection_score_checks(
+            mon_nodes, rados_obj, mon_nums, mon_election_obj
+        ):
+            raise Exception("Connection score checks failed")
+
+        # Changing strategy to 2. i.e. disallowed mode
+        # and perform connection score validation
+        if not mon_election_obj.set_election_strategy(mode="disallow"):
```

Comment:
All the workflows need to be performed in all the election strategies.

Let's iterate through the election modes [classic, disallow , connectivity] and perform all the steps in order.

Currently, most of the scenarios would be performed only in connectivity mode.